sycl/source/detail/queue_impl.cpp

```diff
@@ -609,12 +608,11 @@ EventImplPtr queue_impl::submit_kernel_direct_impl(
         KData.getNDRDesc(), std::move(HostKernelPtr),
         nullptr, // Kernel
         nullptr, // KernelBundle
-        std::move(CGData), std::move(KData).getArgs(),
```

Comment:
why did you remove the `std::move(KData).getArgs()`?

Reply:
`KData` is moved in the following arguments. That's a use-after-move.